lxd/device/device_utils_generic.go

```diff
@@ -16,8 +16,13 @@ func deviceJoinPath(parts ...string) string {
 // validatePCIDevice returns whether a configured PCI device exists. It also returns true, if no device
 // has been specified.
 func validatePCIDevice(config deviceConfig.Device) error {
```

Comment:
@stgraber I think that now the `pci` field is not the only field that can store a PCI device address, and that `address` is generic and could be used by other devices for other meaning, it would be better to alter this function to just take a string argument of `pciAddress` and then it would just be a way of avoiding repetitive calls to `!shared.PathExists(fmt.Sprintf("/sys/bus/pci/devices/%s", pciAddress)))`.

This way we are keeping the device specific logic in the per-device structs and not in the generic helper functions.

Then we would update the devices that currently use `validatePCIDevice` to extract their device specific pci address field (and where relevant skip the step if the device specific address field is empty) and pass it into this function. An additional reason for this is that the existing users of this function the `pci` field is optional, whereas this new device requires the field, so we should avoid having the validation environment skip empty fields (although the config validation should have picked up on this anyway).

Also as its a generic PCI helper, it might be good to move this into the `pci` package as `pcidev.DeviceExists()`.